src/builder.rs

```diff
@@ -911,11 +901,12 @@ impl<'a, 'gcc, 'tcx> BuilderMethods<'a, 'tcx> for Builder<'a, 'gcc, 'tcx> {
     }
 
     fn ptrtoint(&mut self, value: RValue<'gcc>, dest_ty: Type<'gcc>) -> RValue<'gcc> {
-        self.cx.ptrtoint(self.block.expect("block"), value, dest_ty)
+        let usize_value = self.cx.const_bitcast(value, self.cx.type_isize());
+        self.intcast(usize_value, dest_ty, false)
     }
 
     fn inttoptr(&mut self, value: RValue<'gcc>, dest_ty: Type<'gcc>) -> RValue<'gcc> {
-        self.cx.inttoptr(self.block.expect("block"), value, dest_ty)
+        self.cx.const_bitcast(value, dest_ty)
```

Comment:
Perhaps we should add an assert to make sure `value.get_type() == self.isize_type`.

Reply:
Looks like the type doesn't have to be isize for the inttoptr used by the atomic_load and atomic_store intrinsics, so I changed this to do an intcast first like ptrtoint.